spec/services/usps_in_person_proofer_spec.rb

```diff
@@ -0,0 +1,203 @@
+require 'rails_helper'
+
+RSpec.describe UspsInPersonProofer do
+  let(:subject) { UspsInPersonProofer.new }
+
+  describe '#retrieve_token!' do
+    it 'sets token and token_expires_at' do
+      stub_request_token
+      subject.retrieve_token!
+
+      expect(subject.token).to be_present
+      expect(subject.token_expires_at).to be_present
+    end
+  end
+
+  describe '#request_facilities' do
+    it 'returns facilities' do
+      stub_request_token
+      stub_request_facilities
+      location = double('Location',
+        address: Faker::Address.street_address,
+        city: Faker::Address.city,
+        state: Faker::Address.state_abbr,
+        zip_code: Faker::Address.zip_code,
+      )
+
+      subject.retrieve_token!
+      facilities = subject.request_facilities(location)
+
+      facility = facilities['postOffices'][0]
+      expect(facility['distance']).to be_present
+      expect(facility['streetAddress']).to be_present
+      expect(facility['city']).to be_present
+      expect(facility['phone']).to be_present
+      expect(facility['name']).to be_present
+      expect(facility['zip5']).to be_present
+      expect(facility['state']).to be_present
```

Comment:
WDYT of coming up with our own simple PORO response structs to wrap the responses from USPS? Just have this method return an array of `PostOfficeLocation` objects with attrs, then we could rename `zip5` to just `zip` and stuff, `streetAddress` camel case could be `street_address`, etc?

Reply:
good call, done in 9b8ad3b7b